Format UnfoldSim docstrings according to template

jschepers opened the pull request:
As pointed out by @cbrnr (#106), some of the docstrings are still incomplete and do not follow a common docstring convention. @cbrnr thanks a lot for pointing this out and for your suggestions on how to make the docstrings more consistent.

The goal is to transform the docstrings such that they match the docstring template, that I came up with based on the [Julia manual](https://docs.julialang.org/en/v1/manual/documentation/), the [Blue style guide](https://github.com/JuliaDiff/BlueStyle?tab=readme-ov-file#documentation) and by checking the docstrings of other Julia packages.

I will post the current templates for `Functions` and `Types` below. However, please note that they are still subject to change when we find something that does not match the template.

@maanikmarathe, here is a list of all UnfoldSim.jl `src` files with docstrings. Once we are done with them, we can check the boxes. Also, feel free to post about docstrings for which you are unsure how to format them or where information is missing, either in this issue (#108) or by writing to me on Zulip. As discussed, I started from the top of the list and would like you to start from the bottom.

- [x] bases.jl (@jschepers)
- [x] component.jl (@jschepers)
- [x] design.jl (@jschepers)
- [x] headmodel.jl (@jschepers)
- [x] helper.jl (@jschepers)
- [x] noise.jl (@jschepers)
- [x] onset.jl (@jschepers)
- [x] predefinedSimulations.jl (@jschepers)
- [x] simulation.jl (@maanikmarathe)
- [x] types.jl (@jschepers)

jschepers commented on src/simulation.jl:
```diff
@@ -14,28 +14,88 @@ end
+- `signal` : Generated signal. Depending on the design, on the components and on 
+    `return_epoched`, the output can be a 1-D, 2-D, 3-D or 4-D Array. 
+    For example, a 4-D Array would have the dimensions `channels x time x trials x subjects`.
+- `events`: Generated events.
```
```suggestion
- `events`: Generated event table with latencies.
```

jschepers commented on src/simulation.jl:
````diff
@@ -14,28 +14,88 @@ end
+# Examples
+Adapted from the quickstart tutorial in the UnfoldSim docs.
+```julia-repl
+julia> using UnfoldSim
````
For the examples, you can omit the `using UnfoldSim` since the examples are for UnfoldSim functions.

jschepers commented on src/simulation.jl:
```diff
@@ -14,28 +14,88 @@ end
+- `events`: Generated events.
+
+# Examples
+Adapted from the quickstart tutorial in the UnfoldSim docs.
```
Here you could put a link to the quickstart tutorial.

behinger commented on src/onset.jl:
```diff
@@ -1,22 +1,57 @@
-Log-normal inter-event distances using the `Distributions.jl` truncated LogNormal distribution.
+    LogNormalOnset <: AbstractOnset
+
+Log-normal inter-event distances (in samples) using the `Distributions.jl` truncated LogNormal distribution.
```
```suggestion
Log-normal inter-event distances (in samples) using the `Distributions.jl` truncated LogNormal distribution ([code and mathematical reference](https://juliastats.org/Distributions.jl/stable/univariate/#Distributions.LogNormal).
```

behinger commented on src/predefinedSimulations.jl:
```diff
@@ -1,42 +1,85 @@
-- `conditions = Dict(...)`,
+## Design
+- `n_repeats = 100`: Number of times the experimental design is repeated. Only used in the single-subject case.
+- `event_order_function = shuffle`: Random trial order.
```
```suggestion
- `event_order_function = shuffle`: Random trial order. Use `event_order_function = (rng,x)->x` to deactivate
```

behinger commented on src/predefinedSimulations.jl:
```diff
@@ -1,42 +1,85 @@
+## Design
+- `n_repeats = 100`: Number of times the experimental design is repeated. Only used in the single-subject case.
+- `event_order_function = shuffle`: Random trial order.
+- `conditions = Dict(:condition => ["car", "face"], :continuous => range(-5, 5, length = 10))`: Conditions and covariates used in the design.
```
```suggestion
- `conditions = Dict(:condition => ["car", "face"], :continuous => range(-5, 5, length = 10))`: Conditions and covariates used in this predefined design.
```

behinger commented on src/predefinedSimulations.jl:
```diff
@@ -1,42 +1,85 @@
+## Component / Signal
+- `sfreq = 100`: Sampling frequency.
+- `p1 = (p100(; sfreq = sfreq), @formula(0 ~ 1), [5], Dict())`: P1 with amplitude 5; no effects.
+- `n1 = (n170(; sfreq = sfreq), @formula(0 ~ 1 + condition), [5,-3], Dict())`: N1 with amplitude 5, dummy-coded condition effect (levels "car", "face") of -3.
```
because N170 is negative, does this actually mean that face trials are more positive than car trials?

jschepers replied:
Oh yes you are right. I think that's the mistake that we also had in one of the tutorials. I will replace the the -3 by a 3. Or would you prefer to have 2 as an intercept then to have the same values but switched?

behinger replied:
in the code it is also -3; so maybe just the reference category needs to change? I dont want a breaking change on this, but I really dont want to have something where faces are more positive than cars :D

behinger commented on src/predefinedSimulations.jl:
```diff
@@ -1,42 +1,85 @@
 
-#### Onset
+## Onset
 - `overlap = (0.5,0.2)`, # offset + width/length of Uniform noise. put offset to 1 for no overlap. put width to 0 for no jitter
```
```suggestion
- `overlap = (0.5,0.2)`, # convenient parameterization for the default `onset::UniformOnset`. (`offset`, `width`) in seconds. If you do not want any overlap, either use `onset=NoOnset()`, or put the offset to a value larger than any one ERP, e.g. `overlap=(1,0.2)`. Put the `width` to `0` to have no jitter between events.
```

jschepers replied:
I would replace " to a value larger than any one ERP" with "to a value larger than the maximum used component length".

behinger replied:
yes please!

behinger commented on src/predefinedSimulations.jl:
```diff
@@ -139,33 +183,81 @@ function predef_eeg(
-- `contrasts = Dict(:A => EffectsCoding(), :B => EffectsCoding())` # Effect coding by default
-- `formula = n_subjects == 1 ? @formula(0 ~ 1 + A*B) : @formula(dv ~ 1 + A*B + (A*B|subject) + (1|item))`,
+The most used keyword argument is: `return_epoched = true` which returns already epoched data. If you want epoched data without overlap, specify `onset = NoOnset()` and `return_epoched = true`. \n
+Be careful if you modify `n_items` with `n_subjects = 1`, `n_items` has to be a multiple of 4 (or your equivalent conditions factorial, e.g. all combinations length).
```
```suggestion
Be careful if you modify `n_items` with `n_subjects = 1`, `n_items` has to be a multiple of 4 (or your equivalent conditions factorial, e.g. all combinations length).

In difference to `predef_EEG`, `predef_2x2` is sample based (no sampling rate to be specified), and also has a 2x2 design, instead of a 2-categorical, 1-continuous design.
```

jschepers commented on src/simulation.jl:
```diff
@@ -165,12 +282,47 @@ end
+
+# Arguments
+- `signal`: Continuous EEG signal to be modified in place. Has the dimensions `channels x continuous_time x subjects`.
+- `responses::Vector/Matrix/AbstractArray`: Responses to be added, has the dimensions `maxlength(components) x length(simulation.design)`. The data for all the subjects and their respective trials is concatenated.
```
```suggestion
- `responses::Union{Vector, Matrix, AbstractArray}`: Responses to be added. In the multi-channel case, the dimensions are `channels x maxlength(components) x length(simulation.design)`, else `maxlength(components) x length(simulation.design)`.  The data for all the subjects and their respective trials is concatenated.
```
- I also encountered a similar case in one of the other docstrings and decided to use a type union (https://docs.julialang.org/en/v1/manual/types/#Type-Unions).
- I added the dimensions for the case when there are multiple channels.

jschepers reviewed (commented):
@maanikmarathe Thanks a lot for revising the docstrings in `simulation.jl`. I only changed some minor things to make them consistent with the other docstrings and I added a few comments on Github. If you have questions about my suggestions please ask them in a comment below the suggestion. Else it would be great if you could commit them.

jschepers commented:
Finally, all docstrings have been revised :partying_face: Thanks a lot to @maanikmarathe and @behinger for the help!!

Along the way, we fixed the following other things:
- We adapted our LiveServer.jl setup to allow live updating of docstrings in the preview  (based on the [instructions in the LiveServer.jl docs](https://juliadocs.org/LiveServer.jl/dev/man/functionalities/#Issue-with-LiveServer-not-Updating-Documenter-Files-after-a-Change-in-a-Function's-Docstring)).

- Internal variable renaming in `simulation.jl` because some names were ambiguous (#137)
- Breaking change: Revising the `magnitude` function (#132)
- Breaking change: In the `predef_eeg` function the default parameters were adapted such that in the n170 faces are now more negative than cars (which is more realistic).
- Breaking change: Removed `convert` function since it was obsolete in UnfoldSim.jl.

@behinger Did I forget to mention anything? Otherwise, I'd merge the PR to `main`.
